app/services/submissions_expiration_handler.rb

```diff
@@ -0,0 +1,29 @@
+class SubmissionsExpirationHandler
+  def initialize(submissions_inviter: SubmissionsInviter.new)
+    @submissions_inviter = submissions_inviter
+  end
+
+  def call
+    expire_submissions
+    invite_submissions
+  end
+
+  private
+  attr_reader :submissions_inviter
+
+  def expire_submissions
+    submissions_to_be_expired.each do |s|
+      s.expired!
+    end
+  end
+
+  def submissions_to_be_expired
+    Submission.select do |submission|
+      submission.past_confirmation_due_date?
+    end
+  end
```

Comment:
personally, I would probably move expiring to the separate service, to be consistent with `SubmissionsInviter` and just use `submisions_expirer.call` and `submissions_inviter.call` in this service `call` method.

But for sure I would move retrieving submissions to be expired to the repository and use database for that.